src/pip/_internal/req/__init__.py

```diff
@@ -53,40 +59,84 @@ def install_given_reqs(
             ', '.join([req.name for req in to_install]),
         )
 
-    installed = []
+    # pre allocate installed package names
+    installed = [None] * len(
+        to_install
+    )  # type: List[Union[None, InstallationResult, BaseException]]
+
+    install_args = [install_options, global_options, args, kwargs]
 
     with indent_log():
-        for requirement in to_install:
-            if requirement.should_reinstall:
-                logger.info('Attempting uninstall: %s', requirement.name)
-                with indent_log():
-                    uninstalled_pathset = requirement.uninstall(
-                        auto_confirm=True
-                    )
+        if to_install and Pool is not None:
+            # first let's try to install in parallel,
+            # if we fail we do it by order.
+            pool = Pool()
```

Comment:
As per GH-8161, the initialization of `Pool` might fail if `sem_open` is lacking.  `ImportError` is raised during lazy import, so an `try`-`except` might be better guarding.  I am unsure if there's any other errors that can be raised too though.

Reply:
Hmm, good point, I was not aware it will actually fail during the Pool initialization. 
I'll add a guard, and so if it fails, it will basically skip the Pool part.